addons/datatable/scripts/datatable_editor.gd

```diff
@@ -0,0 +1,358 @@
+	
+	var table_panel = Panel.new()
+	table_panel.size_flags_vertical = Control.SIZE_EXPAND_FILL
+	main_vbox.add_child(table_panel)
+	
+	none_lbl = Label.new()
+	none_lbl.text = "Select a datatable resource to edit it here."
+	none_lbl.layout_mode = 1
+	none_lbl.anchors_preset = Control.PRESET_CENTER
+	table_panel.add_child(none_lbl)
+	
+	empty_lbl = Label.new()
+	empty_lbl.text = "No rows in datatable."
+	empty_lbl.layout_mode = 1
+	empty_lbl.anchors_preset = Control.PRESET_CENTER
+	table_panel.add_child(empty_lbl)
+	
+	var table_scroll = ScrollContainer.new()
+	table_scroll.layout_mode = 1
+	table_scroll.anchors_preset = Control.PRESET_FULL_RECT
+	table_panel.add_child(table_scroll)
+	
+	grid_container = GridContainer.new()
+	grid_container.size_flags_horizontal = Control.SIZE_EXPAND_FILL
+	grid_container.size_flags_vertical = Control.SIZE_EXPAND_FILL
+	table_scroll.add_child(grid_container)
+
+func refresh_table():
+	reset_table()
+	populate_table()
+
+func reset_table():
+	for child in grid_container.get_children():
+		child.queue_free()
+	datatable_name_lbl.text = "No Datatable Selected"
+	row_count_lbl.text = "(0 rows)"
+
+func populate_table():
+	if not current_dt:
+		none_lbl.visible = true
+		empty_lbl.visible = false
+		row_count_lbl.visible = false
+		row_type_lbl.visible = false
+		refresh_btn.disabled = true
+		add_row_btn.disabled = true
+		grid_container.visible = false
+		return
+	
+	var row_properties = DatatableUtils.get_row_properties(current_dt.default_row)
+	var row_name = DatatableUtils.get_row_name(current_dt.default_row)
+	var row_count = current_dt.data.size()
+	
+	none_lbl.visible = false
+	empty_lbl.visible = current_dt.data.is_empty()
+	row_count_lbl.visible = true
+	row_type_lbl.visible = true
+	refresh_btn.disabled = false
+	add_row_btn.disabled = false
+	grid_container.visible = true
+	grid_container.columns = row_properties.size() + 3
+	datatable_name_lbl.text = current_dt.resource_path
+	row_count_lbl.text = "(%d row%s)" % [row_count, "s" if row_count != 1 else ""]
+	row_type_lbl.text = "[%s]" % row_name
+	
+	if row_count <= 0:
+		return
+	
+	grid_container.add_child(Control.new())
+	var key_lbl = Label.new()
+	key_lbl.text = "key"
+	grid_container.add_child(key_lbl)
+	for property in row_properties:
+		var header_lbl = Label.new()
+		header_lbl.text = property.name
+		grid_container.add_child(header_lbl)
+	grid_container.add_child(Control.new())
+	
+	var row_index = 0
+	for key in current_dt.data:
+		row_index += 1
+		populate_row(row_index, key)
+
+func populate_row(index: int, key):
+	var idx_lbl = Label.new()
+	idx_lbl.text = " %s " % index
+	idx_lbl.add_theme_color_override("font_color", Color(1, 1, 1, 0.5))
+	idx_lbl.horizontal_alignment = HORIZONTAL_ALIGNMENT_CENTER
+	grid_container.add_child(idx_lbl)
+	
+	var row = current_dt.data[key]
+	var row_properties = DatatableUtils.get_row_properties(row)
+	
+	var key_setter_callback = func(k):
+		match(current_dt.key_type):
+			TYPE_STRING:
+				k = String(k)
```

Comment:
Not a huge fan of single letter variables. Would appreciate more descriptive variable names.